src/main/java/adam/Parser.java

```diff
@@ -0,0 +1,101 @@
+package ada;
+
+import ada.command.AddCommand;
+import ada.command.Command;
+import ada.command.DeleteCommand;
+import ada.command.ExitCommand;
+import ada.command.FindCommand;
+import ada.command.ListCommand;
+import ada.command.MarkCommand;
+import ada.command.UnmarkCommand;
+
+/**
+ * Parses the user input and creates the commands requested by the user.
+ */
+public class Parser {
+    /**
+     * Returns an instance of a Parser object to parse the user input.
+     */
+    public Parser() {
+    }
+
+    /**
+     * Returns the command object after parsing through the user input.
+     *
+     * @param fullCommand The user input.
+     * @return The command object created.
+     * @throws AdaException If the user input is an invalid command or format.
+     */
+    public static Command parse(String fullCommand) throws AdaException {
+        String command = fullCommand.split(" ")[0];
+        String[] splitCommand;
+        String[] splitDesc;

```

Comment:
Technically the variables here should be in plural form since they are of type `String[]` but I can't think of a suitable replacement name either😅.